Foundation/NSError.swift

```diff
@@ -288,7 +288,8 @@ public extension Error where Self : CustomNSError {
 public extension Error {
     /// Retrieve the localized description for this error.
     var localizedDescription: String {
-        return NSError(domain: _domain, code: _code, userInfo: nil).localizedDescription
+        let defaultUserInfo = _swift_Foundation_getErrorDefaultUserInfo(self) as! [String : Any]
```

Comment:
I think this should use as? rather than as! because if it's not a [String:Any] then the effect of passing it in will be nil anyway, and that way, you aren't in danger of a further crash in the error handling.